types/log.go

```diff
@@ -234,3 +234,69 @@ func resolveToHashes(val interface{}) ([]Hash, error) {
 
 	return nil, errors.Errorf("failed to convert %v to hash or hashes", val)
 }
+
+type SubscriptionLog struct {
+	*Log
+	*ChainReorg
+}
+
+type rlpEncodableSubscriptionLog struct {
+	Log        *rlpEncodableLog        `rlp:"nil"`
+	ChainReorg *rlpEncodableChainReorg `rlp:"nil"`
+}
+
+func (s SubscriptionLog) IsRevertLog() bool {
+	return s.ChainReorg != nil
+	// return !reflect.DeepEqual(s.ChainReorg, ChainReorg{})
+}
+
+func (s SubscriptionLog) MarshalJSON() ([]byte, error) {
+	if s.IsRevertLog() {
+		return json.Marshal(s.ChainReorg)
+	}
+	return json.Marshal(s.Log)
+}
```

Comment:
I'm wondering how the SubscribeLogs as following decodes the encoded json string? 

// SubscribeLogs subscribes all logs matching a certain filter, in order.
func (client *Client) SubscribeLogs(channel chan types.SubscriptionLog, filter types.LogFilter) (*rpc.ClientSubscription, error) {
	return client.rpcRequester.Subscribe(context.Background(), "cfx", channel, "logs", filter)
}